src/main/java/Mel.java

```diff
@@ -0,0 +1,204 @@
+import java.io.File;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import task.Todo; // Import the Todo class
+import task.Deadline; // Import the Deadline class
+import task.Event; // Import the Event class
+
+public class Mel {
+
+    public static final String DRAW_HORIZONTAL_LINE = "\t________________________________________";
+    public static final int MARK_WORD_LEN = 4;
+    public static final int UNMARK_WORD_LEN = 6;
+    public static final int INPUT_SPACE_BUFFER = 2;
+    private static final int DELETE_WORD_LEN = 6;
+    private static final String LIST_FILE_PATH = ".\\data\\Mel.txt";
+
+    private static void writeToFile(String filePath, String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    public static void main(String[] args) throws IOException {
+        printIntroMessage();
+        
+        // Set up scanner for user input
+        Scanner in = new Scanner(System.in);
+        List userList = new List();
+
+        try {
+            writerSetUp(LIST_FILE_PATH);
+            loadDataFromFile(LIST_FILE_PATH, userList); // Load saved tasks
+        } catch (IOException e) {
+            System.out.println("An error occurred when setting up writer.");
+        }
+
+        getUserInput(in, userList);
+    }
+
+    private static void writerSetUp(String filePath) throws IOException {
+        File listFile = new File(filePath);
+        if (!listFile.exists()) {
+            File directory = listFile.getParentFile();
+            if (!directory.exists()) {
+                directory.mkdirs(); // Create directory if it doesn't exist
+            }
+            listFile.createNewFile(); // Create file if it doesn't exist
+        }
+    }
+
+    public static void loadDataFromFile(String filePath, List userList) {
+        File file = new File(filePath);
+        try {
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                parseAndAddItem(line, userList);
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found: " + filePath);
+        }
+    }
+
+    private static void parseAndAddItem(String line, List userList) {
+        String[] parts = line.split(" \\| ");
+        String taskType = parts[0]; // T, D, E
+        boolean isDone = parts[1].equals("X");
+        String taskDescription = parts[2];
+
```

Comment:
Add checks to ensure that the number of fields matches what’s expected. If not, log an error and skip the line.